### PR TITLE
Rewrite grouping queries

### PR DESCRIPTION
**Pull request by petroav**
The re-write of the queries will fix some outstanding failures of TPC-DS queries in our benchmarks. See the commit message for a justification on the re-write. Let me know if you disagree.

### CODE REVIEW

File: presto-product-tests/src/main/resources/sql-tests/testcases/tpcds/q36.result [hunk 1]
```diff
@@ -1,101 +1,101 @@
--- delimiter: |; types: DECIMAL|LONGNVARCHAR|CHAR|INTEGER|BIGINT
```
Comment (petroav) [inline presto-product-tests/src/main/resources/sql-tests/testcases/tpcds/q36.result:1]: @kokosing do you remember why this is left over here? Some columns are defined as CHAR but others are still on LONGNVARCHAR?

Comment (kokosing) [reply presto-product-tests/src/main/resources/sql-tests/testcases/tpcds/q36.result:1]: Probably it was quarantined and so it was unable to test. it was left in this shape.

**Review by kokosing — CHANGES_REQUESTED** [review]
Second commit is ready to go.

In regards to first:
 - can you please point to specification where it says that functions on output are disallowed in ORDER BY
 - so far Presto was able to evaluate such functions so I see no reason why you should do an exception
 - if the same expression is used in output and ORDER BY, the it does not have to be evaluated once more, it can be reused

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
lgtm after offline discussion with @sopel39

**Comment by petroav** [comment]
Just to give you @kokosing and @sopel39 more context I looked at the relevant query templates and they all have the alias in the ORDER BY clause instead of the original GROUPING expression.
```
 order by
   lochierarchy desc
  ,case when lochierarchy = 0 then s_state end
  ,rank_within_parent
 [_LIMITC];
```

That is part of the template that is not meant to be substituted by the query generator. I generated the query with 2.4 and 2.1 versions of the generator and I did not get queries with GROUPING in the ORDER BY clause. So I'm essentially reverting these queries back to their original form. I will re-write the first commit message to reflect the intent of the change and will merge once presto-checks passes.

**Comment by petroav** [comment]
Merged to the release branch, thanks!

**Pull request closed without merging**